### PR TITLE
Added some changes for #25 NeworExisting Application Gateway Public IPAddress Pattern in createUidefinition.json and mainTemplate.json files 

### PR DESCRIPTION
**Pull request by Nishu-Dhami**
Working on it.

### CODE REVIEW

**Review by robe070 — CHANGES_REQUESTED** [review]
I can't see where you are declaring `agpublicIpNeworExisting` as a parameter in the template and in the UI. And why is there a 'None'? And the new prompts for existing Public IP Address and DNS should be conditioned on the UI value of `agpublicIpNeworExisting`

File: lansa-vmss-windows-autoscale-sql-database/createUiDefinition.json [hunk 1]
```diff
@@ -591,6 +591,44 @@
...
+                            "hideExisting": false
+                        }
+                    },
+                    {
+                        "name": "Error1",
+                        "type": "Microsoft.Common.InfoBox",
+                        "visible": "[and(or(equals(steps('applicationGatewayConfig').applicationGatewaySkuTier, 'Standard_v2'), equals(steps('applicationGatewayConfig').applicationGatewaySkuTier, 'WAF_v2')), equals(steps('applicationGatewayConfig').publicIpAddress.sku, 'Basic'))]",
+                        "options": {
+                        "icon": "Error",
+                        "text": "For v2 tiers, please create Public IP Address with SKU as Dynamic."
```
Comment (robe070) [inline lansa-vmss-windows-autoscale-sql-database/createUiDefinition.json]: I think that should be "with SKU as Standard"

File: lansa-vmss-windows-autoscale-sql-database/createUiDefinition.json [hunk 2]
```diff
@@ -622,36 +660,66 @@
...
+                            "regex": "^(?!\\s*$).+",
+                            "validationMessage": "The value must not be empty."
                         },
+                        "visible": "[equals(steps('applicationGatewayConfig').autoScalingEnabled, true)]"
+                    },
+                    {
+                        "name": "autoScaleMaxCapacity",
+                        "type": "Microsoft.Common.TextBox",
+                        "label": "Maximum scale units",
+                        "toolTip": "Upper bound on number of Application Gateway capacity",
```
Comment (robe070) [inline lansa-vmss-windows-autoscale-sql-database/createUiDefinition.json]: "Upper bound on Application Gateway capacity"

**Review by robe070 — CHANGES_REQUESTED** [review]
Please answer the comments.
And fix the check thats failed - see below...

**Comment by Nishu-Dhami** [comment]
> Have you created a stack successfully with these changes?
> Please answer the comments

No, The testing part is not yet done. I'm working on it.

File: lansa-vmss-windows-autoscale-sql-database/createUiDefinition.json (hunk 2 above)
Comment (Nishu-Dhami) [reply lansa-vmss-windows-autoscale-sql-database/createUiDefinition.json]: Will add it.

**Comment by robe070** [comment]
It seems like this PR is in progress so I'll be ignoring your commits until you say its ready for me to look at them. Inform me if thats not the case.

**Comment by ashutosh-kumar1** [comment]
@robe070 , The final changes are done and review comments are incorporated. You can review the code. Nishu only needs to test the template with different JSON which she will do today. Will let you know the merge status by today EOD.

**Review by robe070 — APPROVED** [review]

**Comment by ashutosh-kumar1** [comment]
@robe070, the template needs more changes for deployment using an existing public IP address. We will notify you when the final changes are done and tested. The changes can be merged thereafter.

**Pull request merged**